core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -289,8 +294,11 @@ protected ManifestContent content() {
   static class V1Writer extends ManifestWriter<DataFile> {
     private final V1Metadata.IndexedManifestEntry entryWrapper;
 
-    V1Writer(PartitionSpec spec, OutputFile file, Long snapshotId) {
+    V1Writer(PartitionSpec spec, EncryptedOutputFile file, Long snapshotId) {
       super(spec, file, snapshotId);
+      Preconditions.checkState(
+          file.keyMetadata() == null || file.keyMetadata() == EncryptionKeyMetadata.EMPTY,
```

Comment:
Rather than checking identity with `EMPTY` how about checking `file.keyMetadata().buffer() == null`?

Reply:
Per the other comment, we'll remove this limitation